grpc.go

```diff
@@ -29,98 +26,53 @@ func serveGRPC(listener net.Listener, db *gorm.DB) {
 }
 
 func (s *grpcWharfServer) CreateLogStream(stream v5.Builds_CreateLogStreamServer) error {
-	logReqChan := make(chan *v5.CreateLogStreamRequest, 10)
-	var streamErr error
-	go func() {
-		streamErr = recvLogStreamIntoChan(stream, logReqChan)
-	}()
-
 	var logsInserted uint64
 	for {
-		const bufferSize = 100
-		lines := recvQueuedAtLeastOne(logReqChan, bufferSize)
-		if len(lines) == 0 {
+		line, err := stream.Recv()
+		if err == io.EOF {
 			break
+		} else if err != nil {
+			return err
+		} else if line == nil {
+			log.Warn().Message("Received nil message, skipping.")
+			//return status.Error(codes.InvalidArgument, "received nil message")
+			continue
 		}
-		dbLogs := make([]database.Log, 0, len(lines))
-		for _, line := range lines {
-			if err := line.Timestamp.CheckValid(); err != nil {
-				log.Warn().WithError(err).
-					Message("Received invalid log timestamp, skipping.")
-				continue
-			}
-			if line.BuildID == 0 {
-				log.Warn().Message("Received log with build ID: 0, skipping.")
-				continue
-			}
-			if line.BuildID > math.MaxUint {
-				log.Warn().WithUint64("buildId", line.BuildID).
-					Message("Received too big log build ID, skipping.")
-				return status.Errorf(codes.InvalidArgument,
-					"received build ID is too big: %d (build ID) > %d (max)",
-					line.BuildID, uint(math.MaxUint))
-			}
-			dbLogs = append(dbLogs, database.Log{
-				BuildID:   uint(line.BuildID),
-				Message:   line.Message,
-				Timestamp: line.Timestamp.AsTime(),
-			})
+		if err := line.Timestamp.CheckValid(); err != nil {
+			log.Warn().WithError(err).
+				Message("Received invalid log timestamp, skipping.")
+			continue
 		}
-		log.Debug().WithInt("lines", len(lines)).Message("Received log lines")
-		if len(dbLogs) == 0 {
+		if line.BuildID == 0 {
+			log.Warn().Message("Received log with build ID: 0, skipping.")
 			continue
 		}
-		createdLogs, err := createLogBatch(s.db.WithContext(stream.Context()), dbLogs)
+		if line.BuildID > math.MaxUint {
+			log.Warn().WithUint64("buildId", line.BuildID).
+				Message("Received too big log build ID, skipping.")
+			return status.Errorf(codes.InvalidArgument,
+				"received build ID is too big: %d (build ID) > %d (max)",
+				line.BuildID, uint(math.MaxUint))
+		}
+		createdLog, err := saveLog(s.db.WithContext(stream.Context()),
+			uint(line.BuildID),
+			line.Message,
+			line.Timestamp.AsTime(),
+		)
 		if err != nil {
 			return status.Errorf(codes.Internal, "insert logs: %v", err)
 		}
-		log.Debug().WithInt("created", len(createdLogs)).Message("Inserted logs into database")
-		for _, dbLog := range createdLogs {
-			build(dbLog.BuildID).Submit(response.Log{
-				LogID:     dbLog.LogID,
-				BuildID:   dbLog.BuildID,
-				Message:   dbLog.Message,
-				Timestamp: dbLog.Timestamp,
-			})
-		}
-		logsInserted += uint64(len(createdLogs))
-		time.Sleep(time.Second)
+		log.Debug().WithUint("logId", createdLog.LogID).
+			Message("Inserted logs into database.")
```

Comment:
Only one is inserted now.
```suggestion
			Message("Inserted log into database.")
```